src/analytic_engine/src/table_options.rs

```diff
@@ -476,6 +484,22 @@ impl TableOptions {
         m
     }
 
+    /// Check if the options are valid.
+    pub fn is_valid(&self) -> bool {
```

Comment:
Shall we throw the error message to let the caller/user know where the option is wrong?